h5py/tests/test_dataset.py

```diff
@@ -579,43 +583,50 @@ def test_external(self):
 
         assert dset.external is not None
 
-        # verify file was created, and size is correct
-        import os
-        statinfo = os.stat(ext_file)
-        assert statinfo.st_size == testdata.nbytes
-
-        # verify contents
+        # verify file's existence, size, and contents
         with open(ext_file, 'rb') as fid:
             contents = fid.read()
         assert contents == testdata.tostring()
 
-    def test_external_other(self):
-        """ Test other forms of external lists """
+    def test_name_str(self):
+        """ External argument may be a file name str only """
 
-        shape = (6, 100)
-        ext_file = self.mktemp()
+        self.f.create_dataset('foo', (6, 100), external=self.mktemp())
+
+    @ut.skipIf(pathlib is None, "pathlib module not installed")
```

Comment:
We don't need to handle pathlib being absent now - the minimum Python version we're supporting is 3.6, so we can assume pathlib is always there.